upgrade_guide.md

```diff
@@ -0,0 +1,374 @@
+## Class names
+All classes have been namespaced. The mostly used classes are now as follows:
+ 
+- `Algolia::Client` -> `Algolia::Search::Client`
+- `Algolia::Index` -> `Algolia::Search::Index`
+- `Algolia::AccountClient` -> `Algolia::Account::Client`
+- `Algolia::Analytics` -> `Algolia::Analytics::Client`
+- `Algolia::Insights` -> `Algolia::Insights::Client`
+
+## Initialize the client
```

Comment:
I would go with that phrasing and the `# before` / `# after` comments, so what you have to change directly pops to your eyes https://www.algolia.com/doc/api-client/getting-started/upgrade-guides/php/#client-instantiation